app/src/main/java/com/android/wardrobeManager/NewClothingItemActivity.java

```diff
@@ -1,21 +1,164 @@
+        previousActivity = getIntent().getStringExtra("previousActivity");
+        if (getIntent().getParcelableExtra("newClothingItem") != null) {
+            newClothingItem = getIntent().getParcelableExtra("newClothingItem");
+            color = newClothingItem.getColor();
+        }
+        else {
+            color = getResources().getColor(R.color.red);
+            clothingType = "Shirt";
+        }
+
+        imagePreview = findViewById(R.id.newClothingItemImagePreview);
+        imagePreview.setColorFilter(color, PorterDuff.Mode.OVERLAY);
+
+        int[][] states = new int[][] {
+                new int[] { android.R.attr.state_enabled}, // enabled
+                new int[] {-android.R.attr.state_enabled}, // disabled
+                new int[] {-android.R.attr.state_checked}, // unchecked
+                new int[] { android.R.attr.state_pressed}  // pressed
+        };
+
+        int[] colors = new int[] {color, color, color, color};
+
+        colorStateList = new ColorStateList(states, colors);
+
+        colorPreview = findViewById(R.id.colorPreview);
+        colorPreview.setBackgroundTintList(colorStateList);
+
+        itemTypeSpinner = findViewById(R.id.clothingTypeSpinner);
+        itemTypeSpinner.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
+            @Override
+            public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
+                String selectedSpinnerText = itemTypeSpinner.getSelectedItem().toString();
+                itemSelected(selectedSpinnerText);
+            }
+
+            @Override
+            public void onNothingSelected(AdapterView<?> parent) {}
+        });
+
+        if (getIntent().getParcelableExtra("newClothingItem") != null) {
+            clothingType = ((ClothingItem)getIntent().getParcelableExtra("newClothingItem")).clothingTypeToString();
+            changeSpinnerSelection();
+        }
+        else {
+            clothingType = "Shirt";
+        }
+
+        updateImagePreviewClothingType();
+        updateColor();
+
+    }
+
+    public void colorSelected(View view) {
+        colorStateList = findViewById(view.getId()).getBackgroundTintList();
+        color = colorStateList.getDefaultColor();
+        updateColor();
     }
 
+    public void itemSelected(String clothingType) {
+        this.clothingType = clothingType;
 
+        updateImagePreviewClothingType();
+        updateColor();
+    }
+
+    private void updateImagePreviewClothingType() {
+
+        switch (clothingType) {
+            case "Shirt":
+                imagePreview.setImageResource(R.drawable.gray_shirt);
+                break;
+            case "Shoes":
+                imagePreview.setImageResource(R.drawable.gray_shoes);
+                break;
+            case "Shorts":
+                imagePreview.setImageResource(R.drawable.gray_shorts);
+                break;
+        }
```

Comment:
See if you can do the same thing that I mentioned above where you get the resource ID from the string